app/scripts/directives/editEnvironmentFrom.js

```diff
@@ -0,0 +1,208 @@
+        return sourceItemHandleScope.itemScope.sortableScope.$id === destSortableScope.$id;
+      },
+      orderChanged: function() {
+        ctrl.editEnvironmentFromForm.$setDirty();
+      }
+    };
+
+    ctrl.envFromObjectSelected = function(index, entry, selected) {
+      var newEnvFrom = {};
+
+      switch (selected.kind) {
+        case 'Secret':
+          newEnvFrom.secretRef = {
+            name: selected.metadata.name
+          };
+          delete ctrl.envFromEntries[index].configMapRef;
+          break;
+        case 'ConfigMap':
+          newEnvFrom.configMapRef = {
+            name: selected.metadata.name
+          };
+          delete ctrl.envFromEntries[index].secretRef;
+          break;
+      }
+
+      _.assign(ctrl.envFromEntries[index], newEnvFrom);
+      ctrl.updateEntries(ctrl.envFromEntries);
+    };
+
+    ctrl.updateEntries = function(entries) {
+      ctrl.entries = _.filter(entries, function (val) {
+        return val.secretRef || val.configMapRef;
+      });
+    };
+
+    var updateEnvFromEntries = function(entries) {
+      ctrl.envFromEntries = entries || [];
+
+      if(!ctrl.envFromEntries.length) {
+        addEntry(ctrl.envFromEntries);
+      }
+
+      _.each(ctrl.envFromEntries, function(entry) {
+        if(entry) {
+          if(entry.configMapRef && !canI('configmaps', 'get')) {
+            entry.isReadonlyValue = true;
+          }
+
+          if(entry.secretRef && !canI('secrets', 'get')) {
+            entry.isReadonlyValue = true;
+          }
+        }
+      });
+    };
+
+    var getReferenceValue = function(option) {
+      var referenceValue;
+
+      switch(option.kind) {
+        case 'ConfigMap':
+          referenceValue = _.find(ctrl.envFromEntries, {configMapRef: {name: option.metadata.name}});
+          break;
+        case 'Secret':
+          referenceValue = _.find(ctrl.envFromEntries, {secretRef: {name: option.metadata.name}});
+          break;
+      }
+
+      return referenceValue;
+    };
+
+    ctrl.checkEntries = function(option) {
+      return !!(getReferenceValue(option));
+    };
+
+    var findReferenceValueForEntries = function(entries, envFromSelectorOptions) {
+      ctrl.cannotAdd = (ctrl.isReadonlyAny || _.isEmpty(envFromSelectorOptions));
+
+      if(envFromSelectorOptions) {
+        _.each(envFromSelectorOptions, function(option) {
+          var referenceValue = getReferenceValue(option);
+
+          if (referenceValue) {
+            _.set(referenceValue, 'selectedEnvFrom', option);
+          }
+        });
+
+      } else {
+        _.each(entries, function(entry) {
+          var entryKind = (entry.secretRef)? 'Secret' : 'ConfigMap';
+          var entryName = _.get(entry.configMapRef, 'name') || _.get(entry.secretRef, 'name');
+
+          _.set(entry, 'apiObj', {
+            kind: entryKind,
+            metadata: {
+              name: entryName,
+              namespace: namespace
+            }
+          });
+        });
```

Comment:
I'd rather not set dummy values. Is there another way to handle this? Is this for handling secrets that have been deleted?

Reply:
For the comment chain, we're removing this as discussed